core/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -123,7 +123,9 @@ public synchronized ClientStream newStream(
       return new NoopClientStream() {
         @Override
         public void start(ClientStreamListener listener) {
-          listener.closed(shutdownStatus, new Metadata());
+          synchronized (InProcessTransport.this) {
+            listener.closed(shutdownStatus, new Metadata());
```

Comment:
The GaurdedBy annotation causes a build failure internally without this.   It's actually the shutdownStatus that has the problem.